alacritty/src/event.rs

```diff
@@ -1327,21 +1339,167 @@ impl Processor {
+    ) -> GlutinResult<NotCurrentContext> {
+        let context_attributes = ContextAttributesBuilder::new()
+            .with_context_api(ContextApi::OpenGl(Some(Version::new(3, 3))))
+            .build(raw_window_handle);
+
+        unsafe {
+            if let Ok(gl_context) = gl_display.create_context(gl_config, &context_attributes) {
+                Ok(gl_context)
+            } else {
+                // Fallback.
+                let context_attributes = ContextAttributesBuilder::new()
+                    .with_context_api(ContextApi::OpenGl(Some(Version::new(2, 1))))
+                    .build(raw_window_handle);
+                gl_display.create_context(gl_config, &context_attributes)
+            }
+        }
+    }
+
+    fn pick_gl_config(
+        gl_display: &GlutinDisplay,
+        raw_window_handle: Option<RawWindowHandle>,
+    ) -> Result<GlutinConfig, String> {
+        let mut default_config = ConfigTemplateBuilder::new().with_transparency(true);
+
+        if let Some(raw_window_handle) = raw_window_handle {
+            default_config = default_config.compatible_with_native_window(raw_window_handle);
+        }
+
+        let config_10bit = default_config
+            .clone()
+            .with_buffer_type(ColorBufferType::Rgb { r_size: 10, g_size: 10, b_size: 10 })
+            .with_alpha_size(2);
+
+        let configs = [
+            default_config.clone(),
+            config_10bit.clone(),
+            default_config.with_transparency(false),
+            config_10bit.with_transparency(false),
+        ];
+
+        for config in configs {
+            let gl_config = unsafe {
+                gl_display.find_configs(config.build()).ok().and_then(|mut configs| configs.next())
+            };
+
+            if let Some(gl_config) = gl_config {
+                return Ok(gl_config);
+            }
+        }
+
+        Err(String::from("failed to find suitable GL configuration."))
+    }
+
+    /// Create initial window and load GL platform.
+    ///
+    /// This will initialize the OpenGL Api and pick a config that
+    /// will be used for the rest of the windows.
+    pub fn create_initial_window(
+        &mut self,
+        event_loop: &EventLoopWindowTarget<Event>,
+        proxy: EventLoopProxy<Event>,
+        options: WindowOptions,
+    ) -> Result<(), Box<dyn Error>> {
+        let raw_display_handle = event_loop.raw_display_handle();
+
+        let mut identity = self.config.window.identity.clone();
+        options.window_identity.override_identity_config(&mut identity);
+
+        // Windows has different order of GL platform initialization compared to any other platform,
+        // it requires window first.
+        #[cfg(windows)]
+        let window = Some(Window::new(event_loop, &self.config, &identity)?);
+        #[cfg(not(windows))]
+        let window: Option<Window> = None;
+
+        let raw_window_handle = window.as_ref().map(|w| w.raw_window_handle());
```

Comment:
What's the point of the as_ref().map when it's always going to be None on macos/linux and some on windows anyway? Don't need to create a temporary `window` variable at all.

Reply:
What do you mean? I need to initialize window on windows before doing anything else and keep it around? You mean that I don't need that variable on non windows? Then yes, but it's less code to write that way and compiler will optimize out that `None` parts.

Reply:
I've added just more cfgs here, we'd still need `RWHD` though, so I just assign it to `None`.

Reply:
Yeah I like it better this way.